notebooks/official/ray_on_vertex_ai/spark_on_ray_on_vertex_ai.ipynb

```diff
@@ -0,0 +1,1353 @@
+      },
+      "source": [
+        "### Enable Artifact Registry API\n",
+        "Enable the Artifact Registry API service for the Google cloud project. This tutorial assumes that [gcloud CLI](https://cloud.google.com/sdk/docs/install) is already installed."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "26b48ef4ba29"
+      },
+      "outputs": [],
+      "source": [
+        "! gcloud services enable artifactregistry.googleapis.com"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b37f718f7eae"
+      },
+      "source": [
+        "### Create a private Docker repository\n",
+        "Create a Docker repository in Google Artifact Registry."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0a2f0e3b1cd5"
+      },
+      "outputs": [],
+      "source": [
+        "DOCKER_REPOSITORY = \"my-rov-repo\"\n",
+        "IMAGE_NAME = \"raydp-rov-image\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "34d681acb474"
+      },
+      "outputs": [],
+      "source": [
+        "! gcloud artifacts repositories create {DOCKER_REPOSITORY} --repository-format=docker --location={LOCATION} --description=\"Docker repository\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "93ef2bf0de2d"
+      },
+      "source": [
+        "### Build container image\n",
+        "This tutorial assumes that [Docker](https://docs.docker.com/engine/install/) is installed and available in the work environment."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "413836afe0e9"
+      },
+      "outputs": [],
+      "source": [
+        "!docker build {DOCKER_DIR} -t {LOCATION}-docker.pkg.dev/{PROJECT_ID}/{DOCKER_REPOSITORY}/{IMAGE_NAME}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b8ddd31aa438"
+      },
+      "source": [
+        "### Push container image\n",
+        "Configure the [authentication for Google Artifact Registry's Docker repository](https://cloud.google.com/artifact-registry/docs/docker/pushing-and-pulling#auth) before pushing the container image to the repository."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ce0ecd10af62"
+      },
+      "outputs": [],
+      "source": [
+        "!docker push {LOCATION}-docker.pkg.dev/{PROJECT_ID}/{DOCKER_REPOSITORY}/{IMAGE_NAME}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "57ad7b911d1d"
+      },
+      "source": [
+        "## Create Ray cluster on Vertex AI\n",
+        "Use the custom container image to create Ray cluster on Vertex AI using Vertex AI Python SDK."
```

Comment:
- insert "a": "Use the custom container image to create a Ray cluster on Vertex AI using Vertex AI Python SDK"